crates/orderbook/src/database/orders.rs

```diff
@@ -356,6 +359,68 @@ impl OrderStoring for Postgres {
         .try_collect()
         .await
     }
+
+    async fn order_status(&self, order_uid: &OrderUid) -> Result<Status> {
+        let mut ex = self.pool.begin().await.context("could not init tx")?;
+        let timer = super::Metrics::get()
+            .database_queries
+            .with_label_values(&["latest_order_event"])
+            .start_timer();
+        let status = database::order_events::get_latest(&mut ex, &ByteArray(order_uid.0))
+            .await
+            .context("could not fetch status")?;
+        timer.stop_and_record();
+
+        let fetch_solvers = || async move {
+            let timer = super::Metrics::get()
+                .database_queries
+                .with_label_values(&["load_latest_solver_competition"])
+                .start_timer();
+            let competition = database::solver_competition::load_latest_competition(&mut ex)
```

Comment:
What if the order exists but it is still not part of the latest competition? :thinking:

Reply:
A warning log should probably be enough here. @alfetopito, should we fail in case the solvers list for the order in specific status is unexpectedly empty or can we pass the empty list to you?

Reply:
I don't think a failure is adequate here.
After all, there is a status, just not the full list of solvers is known, right?
I rather have it empty than not have it.

Reply:
Ignoring the competition fetching errors now.